crates/ruff_python_ast/src/traversal.rs

```diff
@@ -0,0 +1,108 @@
+    match parent {
+        Stmt::FunctionDef(ast::StmtFunctionDef { body, .. }) => Some(body),
+        Stmt::AsyncFunctionDef(ast::StmtAsyncFunctionDef { body, .. }) => Some(body),
+        Stmt::ClassDef(ast::StmtClassDef { body, .. }) => Some(body),
+        Stmt::For(ast::StmtFor { body, orelse, .. }) => {
+            if body.contains(stmt) {
+                Some(body)
+            } else if orelse.contains(stmt) {
+                Some(orelse)
+            } else {
+                None
+            }
+        }
+        Stmt::AsyncFor(ast::StmtAsyncFor { body, orelse, .. }) => {
+            if body.contains(stmt) {
+                Some(body)
+            } else if orelse.contains(stmt) {
+                Some(orelse)
+            } else {
+                None
+            }
+        }
+        Stmt::While(ast::StmtWhile { body, orelse, .. }) => {
+            if body.contains(stmt) {
+                Some(body)
+            } else if orelse.contains(stmt) {
+                Some(orelse)
+            } else {
+                None
+            }
+        }
+        Stmt::If(ast::StmtIf { body, orelse, .. }) => {
+            if body.contains(stmt) {
+                Some(body)
+            } else if orelse.contains(stmt) {
+                Some(orelse)
+            } else {
+                None
+            }
+        }
+        Stmt::With(ast::StmtWith { body, .. }) => Some(body),
+        Stmt::AsyncWith(ast::StmtAsyncWith { body, .. }) => Some(body),
+        Stmt::Match(ast::StmtMatch { cases, .. }) => cases
+            .iter()
+            .map(|case| &case.body)
+            .find(|body| body.contains(stmt)),
+        Stmt::Try(ast::StmtTry {
+            body,
+            handlers,
+            orelse,
+            finalbody,
+            ..
+        }) => {
+            if body.contains(stmt) {
+                Some(body)
+            } else if orelse.contains(stmt) {
+                Some(orelse)
+            } else if finalbody.contains(stmt) {
+                Some(finalbody)
+            } else {
+                handlers
+                    .iter()
+                    .filter_map(ExceptHandler::as_except_handler)
+                    .map(|handler| &handler.body)
+                    .find(|body| body.contains(stmt))
+            }
+        }
+        Stmt::TryStar(ast::StmtTryStar {
+            body,
+            handlers,
+            orelse,
+            finalbody,
+            ..
+        }) => {
+            if body.contains(stmt) {
+                Some(body)
+            } else if orelse.contains(stmt) {
+                Some(orelse)
+            } else if finalbody.contains(stmt) {
+                Some(finalbody)
+            } else {
+                handlers
+                    .iter()
+                    .filter_map(ExceptHandler::as_except_handler)
+                    .map(|handler| &handler.body)
+                    .find(|body| body.contains(stmt))
+            }
+        }
+        _ => None,
```

Comment:
Do you think we should bias towards this pattern (explicit enumeration of AST nodes) everywhere? Or use it selectively?

Reply:
I use it selectively based on the use case. I use this pattern when I realize that I start by generating all match cases and remove the ones I don't care about because I can't easily tell which nodes should be handled. 

Another way to think about this. Would an author adding a new statement know that they need to add it here? If not, then it could make sense to use an exhaustive match

Reply:
Makes sense